pkg/nfd-master/updater-pool.go

```diff
@@ -85,16 +90,55 @@ func (u *nodeUpdaterPool) runNodeUpdater(queue workqueue.RateLimitingInterface)
 	u.wg.Done()
 }
 
-func (u *nodeUpdaterPool) start(parallelism int) {
+func (u *updaterPool) processNodeFeatureGroupUpdateRequest(cli nfdclientset.Interface, ngfQueue workqueue.RateLimitingInterface) bool {
+	nfgName, quit := ngfQueue.Get()
+	if quit {
+		return false
+	}
+	defer ngfQueue.Done(nfgName)
+
+	nodeFeatureGroupUpdateRequests.Inc()
+
+	// Check if NodeFeatureGroup exists
+	var nfg *nfdv1alpha1.NodeFeatureGroup
+	var err error
+	if nfg, err = getNodeFeatureGroup(cli, u.nfdMaster.namespace, nfgName.(string)); apierrors.IsNotFound(err) {
+		klog.InfoS("NodeFeatureGroup not found, skip update", "NodeFeatureGroupName", nfgName)
+	} else if err := u.nfdMaster.nfdAPIUpdateNodeFeatureGroup(u.nfdMaster.nfdClient, nfg); err != nil {
+		if ngfQueue.NumRequeues(nfgName) < 15 {
+			klog.InfoS("retrying NodeFeatureGroup update")
+			ngfQueue.AddRateLimited(nfgName)
+			return true
+		} else {
+			klog.ErrorS(err, "failed to update NodeFeatureGroup")
+		}
```

Comment:
Let's retry indefinitely (like we do for nodes), or wdyt(?) Also, a bit more verbose log messages

```suggestion
		if ngfQueue.NumRequeues(nfgName) < 15 {
			klog.InfoS("retrying NodeFeatureGroup update", "nodeFeatureGroup", klog.KObj(nfg), "lastError", err)
		} else {
			klog.ErrorS(err, "failed to update NodeFeatureGroup, queueing for retry", "nodeFeatureGroup", klog.KObj(nfg), "lastError", err, "numRetries", n)
		}
		ngfQueue.AddRateLimited(nfgName)
		return true
```

I think it could be also just one error message, instead of info and err (separated by a rather arbitrary `< 15`), if you like.